enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,373 @@
+from those frameworks try to come up with a solution that we can reuse for new installers.
+
+### Goals
+
+* This proposal will focus using this new e2e framework in the new ephemeral agent based 
+installer image project.
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* There should be some overlap in the pieces that are refactored for use in the new project.
+However it is still a non-goal for outside installer to use this new framework in their e2e 
+tests for the upcoming release. We will plan to create a point of convergence after the
+ephemeral agent based installer project is out of prototyping and MVP phase.
+
+## Proposal
+
+Create an e2e testing framework for the ephemeral agent based installer image in the 
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts) repository and refactor 
+components as needed for the new project.
+
+### User Stories
+
+#### Setup Environment for Install Without Installing OCP or k8s
+
+As a developer, I need to set up VMs so that I can simulate the installation of a users 
+first cluster using the ephemeral based agent installer image.
+
+**Implementation Plan:**
+
+* Use the current shell scripts in dev-scripts to set up the VMs for the agent-installer.
+
+The current scripts we will use in dev-scripts are 
+[01_install_requirements.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/01_install_requirements.sh) 
+and [02_configure_host.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/02_configure_host.sh).
+
+
+* Create new shell scripts to run routines specific to agent installer after VMs are created. An example 
+of this is launching VM with new ISO from agent installer project.
+
+#### Run Assisted Installer Outside k8s
+
+* As a developer, I need to run Assisted Installer without installing a k8s platform first 
+so that I can test the installation of a users first cluster, cluster 0.
+
+* As a developer, I need to launch a VM with the built ephemeral agent based installer
+image so that I can test and validate this image. 
+
+This is the overall purpose of this ephemeral agent based installer image project. 
+
+* The Assisted Installer e2e tests that run in dev-scripts first use the IPI installer 
+to install OCP and then the infrastructure-operator on this OCP cluster making it a hub 
+cluster.
+
+* The Assisted Installer e2e tests that run in assisted-test-infra installs minikube first 
+and then runs Assisted Installer on top of it.
+
+Which means currently in both cases we first setup some form of k8s first. Given
+the goal of the ephemeral agent based installer project to install a users first cluster, 
+cluster 0, this is highly undesirable.
+
+**Implementation Plan:**
+
+* Create a new routine to run Assisted Installer outside of k8s.
+
+* Create a new routine to launch a VM with the built ephemeral agent based installer
+image.
```

Comment:
What about routines to:
1. Check cluster installation progress
2. Check cluster installation is successful
3. Validate kubeconfig is available and it can be used to access newly formed cluster
4. Extract logs

Reply:
I added these as user stories.